tests/Microsoft.Bot.Builder.Tests/Teams/TeamsSSOTokenExchangeMiddlewareTests.cs

```diff
@@ -0,0 +1,201 @@
+        private const string Token = "token";
+
+        [Fact]
+        public void ConstructorValidation()
+        {
+            Assert.Throws<ArgumentNullException>(() => new TeamsSSOTokenExchangeMiddleware(null, ConnectionName));
+            Assert.Throws<ArgumentNullException>(() => new TeamsSSOTokenExchangeMiddleware(new MemoryStorage(), null));
+            Assert.Throws<ArgumentNullException>(() => new TeamsSSOTokenExchangeMiddleware(new MemoryStorage(), string.Empty));
+        }
+
+        [Fact]
+        public async Task TokenExchanged_OnTurnFires()
+        {
+            // Arrange
+            bool wasCalled = false;
+            var adapter = new TeamsSSOAdapter(CreateConversationReference())
+               .Use(new TeamsSSOTokenExchangeMiddleware(new MemoryStorage(), ConnectionName));
+            
+            adapter.AddExchangeableToken(ConnectionName, Channels.Msteams, TeamsUserId, FakeExchangeableItem, Token);
+
+            // Act
+            await new TestFlow(adapter, async (context, cancellationToken) =>
+            {
+                // note the Middleware should not cause the Responded flag to be set
+                Assert.False(context.Responded);
+                wasCalled = true;
+                await context.SendActivityAsync("processed", cancellationToken: cancellationToken);
+                await Task.CompletedTask;
+            })
+                .Send("test")
+                .AssertReply("processed")
+                .StartTestAsync();
+
+            // Assert
+            Assert.True(wasCalled, "Delegate was not called");
+        }
+
+        [Fact]
+        public async Task TokenExchanged_SecondSendsInvokeResponse()
+        {
+            // Arrange
+            bool wasCalled = false;
+            int calledCount = 0;
+            var adapter = new TeamsSSOAdapter(CreateConversationReference())
+               .Use(new TeamsSSOTokenExchangeMiddleware(new MemoryStorage(), ConnectionName));
+
+            adapter.AddExchangeableToken(ConnectionName, Channels.Msteams, TeamsUserId, FakeExchangeableItem, Token);
+
+            // Act
+            await new TestFlow(adapter, async (context, cancellationToken) =>
+            {
+                // note the Middleware should not cause the Responded flag to be set
+                Assert.False(context.Responded);
+                wasCalled = true;
+                calledCount++;
+                await context.SendActivityAsync("processed", cancellationToken: cancellationToken);
+                await Task.CompletedTask;
+            })
+                .Send("test")
+                .AssertReply("processed")
+                .Send("test")
+                .AssertReply((activity) =>
+                {
+                    // When the 2nd message goes through, it is not processed due to deduplication
+                    // but an invokeResponse of 200 status with empty body is sent back
+                    Assert.Equal("invokeResponse", activity.Type);
```

Comment:
ActivityTypesEx.InvokeResponse instead of `"invokeResponse"`

https://github.com/microsoft/botbuilder-dotnet/blob/f7b290450944b18495ab78d833c99f777bae4509/libraries/Microsoft.Bot.Schema/ActivityTypesEx.cs#L26